### PR TITLE
Revise docs on randomness

### PR DESCRIPTION
**Pull request by JoshOrndorff**
This is my attempt to solve #5496 and start working on https://github.com/substrate-developer-hub/knowledgebase/issues/15

I've moved the docs to the places I _think_ they should go, but I'm not confident I've gotten everything right, and would appreciate careful review. In particular I'm not sure whether this warning applies to Collective Flip, Babe, or both
```
	/// WARNING: Hashing the result of this function will remove any low-influence properties it has
	/// and mean that all bits of the resulting value are entirely manipulatable by the author of
	/// the parent block, who can determine the value of `parent_hash`.
```

### CODE REVIEW

**Comment by JoshOrndorff** [comment]
Also requesting review from @burdges

**Comment by JoshOrndorff** [comment]
> paritytech-ci added the B2-breaksapi label

I think that's incorrect

File: frame/randomness-collective-flip/src/lib.rs [hunk 1]
```diff
@@ -98,40 +98,13 @@ decl_storage! {
...
 
 impl<T: Trait> Randomness<T::Hash> for Module<T> {
-	/// Get a low-influence "random" value.
-	///
-	/// Being a deterministic block chain, real randomness is difficult to come by. This gives you
-	/// something that approximates it. `subject` is a context identifier and allows you to get a
-	/// different result to other callers of this function; use it like
-	/// `random(&b"my context"[..])`. This is initially implemented through a low-influence
-	/// "triplet mix" convolution of previous block hash values. In the future it will be generated
-	/// from a secure verifiable random function (VRF).
```
Comment (DemiMarie-parity) [inline frame/randomness-collective-flip/src/lib.rs:108]: Why was this deleted?

Comment (JoshOrndorff) [reply frame/randomness-collective-flip/src/lib.rs:108]: Most of it is already present on the `Randomness` trait itself. Except the part about "In the future..." which was removed because the future is here :)

File: frame/babe/src/lib.rs [hunk 2]
```diff
@@ -195,6 +195,25 @@ decl_module! {
 }
 
 impl<T: Trait> RandomnessT<<T as frame_system::Trait>::Hash> for Module<T> {
+	/// Some BABE blocks have VRF outputs where the block producer has exactly one bit of influence,
+	/// either they make the block or they do not make the block and thus someone else makes the
+	/// next block. Yet, this randomness is not fresh in all BABE blocks.
```
Comment (burdges) [inline frame/babe/src/lib.rs:200]: All BABE blocks need fresh VRF outputs for randomness.  Initially, the extra AURA-BABE blocks lacked these fresh VRF outputs, but parachains depend upon the, so they must be added.

**Comment by bkchr** [comment]
@rphmeier can you review this please?

@JoshOrndorff can you please merge master?

File: frame/babe/src/lib.rs (hunk 2 above)
Comment (andresilva) [reply frame/babe/src/lib.rs:200]: This is still the case as #5501 has not been merged yet. We should update these docs on that PR to reflect the changes.

**Comment by JoshOrndorff** [comment]
I've merged master.
I've also removed the breaks-api tag because I think ci incorrectly added it. Please add it back if it should be there.

From my perspective, there is one remaining open question before this is ready for merge. Does the following warning apply to collective flip, Babe, or all implementors of the `Randomness` trait?
```
/// WARNING: Hashing the result of this function will remove any low-influence properties it has
/// and mean that all bits of the resulting value are entirely manipulatable by the author of
/// the parent block, who can determine the value of `parent_hash`.
```

File: frame/babe/src/lib.rs [hunk 3]
```diff
@@ -205,6 +205,25 @@ decl_module! {
...
+	/// - Name, in advance, the block number whose random value will be used; ensure your module
+	///   retains a buffer of previous random values for its subject and then index into these in
+	///   order to obviate the ability of your user to look up the parent hash and choose when to
+	///   transact based upon it.
+	/// - Require your user to first commit to an additional value by first posting its hash.
+	///   Require them to reveal the value to determine the final result, hashing it with the
+	///   output of this random function. This reduces the ability of a cabal of block producers
+	///   from conspiring against individuals.
+	///
+	/// WARNING: Hashing the result of this function will remove any low-influence properties it has
```
Comment (rphmeier) [inline frame/babe/src/lib.rs:224]: This is copied from `randomness-collective-flip`, isn't it?
I don't believe it is accurate here.

**Comment by burdges** [comment]
We've no worries about hashing randomness from BABE, only randomness from the low influence collective flip scheme.  We should advise that randomness-collective-flip itself exists for low security situations, like tests.

**Comment by rphmeier** [comment]
Thanks for confirming my suspicion @burdges :) I agree that the extra addendum to the `randomness-collective-flip` docs would be good in this PR.

**Comment by JoshOrndorff** [comment]
@rphmeier I've addressed your comments and merged master.
